pkg/ssh/connection_golang.go

```diff
@@ -272,17 +273,38 @@ func ValidateAndConfigure(uri *url.URL, iden string) (*ssh.ClientConfig, error)
 	if err != nil {
 		return nil, err
 	}
-	keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
-	known, err := knownhosts.New(keyFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("error creating host key callback function for %s: %w", keyFilePath, err)
-	}
 
+	var keyErr *knownhosts.KeyError
 	cfg := &ssh.ClientConfig{
-		User:            uri.User.Username(),
-		Auth:            authMethods,
-		HostKeyCallback: known,
-		Timeout:         tick,
+		User: uri.User.Username(),
+		Auth: authMethods,
+		HostKeyCallback: ssh.HostKeyCallback(func(host string, remote net.Addr, pubKey ssh.PublicKey) error {
+			keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
+			known, err := knownhosts.New(keyFilePath)
+			if err != nil {
+				if os.IsNotExist(err) {
+					logrus.Warn("podman will soon require host key verification. This means a known_hosts file must exist in the user's ssh directory and a verified key must be within it")
+					return nil
+				}
+				return err
+			}
+			hErr := known(host, remote, pubKey)
+			if errors.As(hErr, &keyErr) && len(keyErr.Want) > 0 {
+				logrus.Warnf("WARNING: %v of type %v is not a key of %s", ssh.FingerprintSHA256(pubKey), pubKey.Type(), host)
+				return keyErr
+			} else if errors.As(hErr, &keyErr) && len(keyErr.Want) == 0 {
+				logrus.Warn("key not found in known_hosts, podman will soon require keys to exist in known_hosts. If you are using a podman machinr or an old system connection, please recreate using podman machine init or podman system connection add")
+			}
+			return nil
+		}),
+		Timeout: tick,
+		HostKeyAlgorithms: []string{
```

Comment:
Why was this part added?

I’m not at all saying that the values are incorrect or undesirable, at this point — my concern right now is that it would be better to avoid getting into the business of being responsible for maintaining such policy choices _over time_, in the first place. This will quite likely need updating in 5–10 years, and we have no idea who, if anyone, will be around to do that.

Reply:
I think (not sure) one specific key type was giving a lot of trouble with golang's processing. I tried it on multiple machines and without this specification, a key mismatch or a key unknown error occurs due to a type that golang cannot process.